src/components/slider/daikin-slider.ts

```diff
@@ -0,0 +1,299 @@
+      width: 100%;
+    }
+  `;
+
+  /**
+   * The lowest value in the range of permitted values.
+   */
+  @property({ type: String, reflect: true })
+  min = "1";
+
+  /**
+   * The greatest value in the range of permitted values.
+   */
+  @property({ type: String, reflect: true })
+  max = "100";
+
+  /**
+   * The step attribute is a number that specifies the granularity that the value must adhere to.
+   */
+  @property({ type: String, reflect: true })
+  step = "1";
+
+  /**
+   * The form name, submitted as a name/value pair when submitting the form.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * The form value, submitted as a name/value pair when submitting the form.
+   */
+  @property({ type: String, reflect: true })
+  value = "1";
+
+  /**
+   * Specify the slider disabled state.
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  static readonly formAssociated = true;
+
+  private _sliderRef: Ref<HTMLElement> = createRef();
+
+  // define _internals to let the slider can be used in a form
+  private _internals = this.attachInternals();
+
+  private _updateFormValue() {
+    if (this.disabled) {
+      return;
+    }
+    this._internals.setFormValue(this.value);
+    this.dispatchEvent(
+      new Event("change", {
+        bubbles: true,
+        composed: true,
+      })
+    );
+    this.dispatchEvent(
+      new Event("input", {
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  private _updateProgressFromValue() {
+    const min = parseFloat(this.min);
+    const max = parseFloat(this.max);
+    const valueCalc = Math.max(min, Math.min(parseFloat(this.value), max));
+    return ((valueCalc - min) / (max - min)) * 100;
+  }
+
+  // This function will triggered when click the slider bar area.
+  private _handleClick(event: MouseEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    const leftDistance = this.getThumbFraction(event);
+    if (leftDistance == null) {
+      return;
+    }
+    const value = getValueAndProgressFromCoordinate(this, leftDistance);
+    this.value = value;
+  }
+
+  // This function will triggered when the slider thumb button be focused and click keyboard arrow key.
+  private _handleKeyDown(event: KeyboardEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    if (event.key === "Home") {
+      this.value = this.min;
+      return;
+    }
+    if (event.key === "End") {
+      this.value = this.max;
+      return;
+    }
+    const step = parseFloat(this.step);
+    const moveOffset = {
+      ArrowRight: step,
+      ArrowDown: -step,
+      ArrowLeft: -step,
+      ArrowUp: step,
+    }[event.key];
+    if (!moveOffset) {
+      return;
+    }
+    const value = getValueAndProgressFromKeyboard(this, moveOffset);
+    this.value = value;
+  }
+
+  // Get the position of the current thumb icon as a fraction of the entire slider bar.
```

Comment:
```suggestion
  /**
   * Returns the normalized position of the mouse on the slider, within the range 0 to 1.
   *
   * @param event A MouseEvent.
   * @returns The mouse's X coordinate normalized to a range of 0 to 1. The left end and beyond of the slider is 0, and the right end and beyond is 1.
   */
```

The position of the thumb is controlled by `value`, so it is not correct to use that word here.
The process is carried out in the following way: 1) `value` is updated based on the mouse position, and 2) the position of the thumb is updated as a result. This function is used in the first stage, so the "mouse position" is correct.